src/validation.cpp

```diff
@@ -2507,7 +2510,7 @@ bool CChainState::ActivateBestChainStep(BlockValidationState& state, CBlockIndex
         if (!DisconnectTip(state, &disconnectpool)) {
             // This is likely a fatal error, but keep the mempool consistent,
             // just in case. Only remove from the mempool in this case.
-            UpdateMempoolForReorg(*this, m_mempool, disconnectpool, false);
+            if (m_mempool) UpdateMempoolForReorg(*this, *m_mempool, disconnectpool, false);
```

Comment:
What do you think about updating `UpdateMempoolForReorg` to either be a member of `CChainState` or take `CTxMemPool` as an optional pointer, and then return immediately if mempool is null. That would avoid scattering so many `if (m_mempool)` checks across multiple validation functions.

Reply:
I went ahead and threw it onto `CChainState` since I think that makes sense; can strip off the commit if there are any objections.